tests/FSharp.Test.Utilities/ProjectGeneration.fs

```diff
@@ -792,12 +794,14 @@ type WorkflowContext =
       Signatures: Map<string, string>
       Cursor: FSharpSymbolUse option }
 
-let SaveAndCheckProject project checker =
+let SaveAndCheckProject project checker isExistingProject =
     async {
         use _ =
             Activity.start "SaveAndCheckProject" [ Activity.Tags.project, project.Name ]
 
-        do! saveProject project true checker
+        // Don't save the project if it is a real world project that exists on disk.
+        if not isExistingProject then
+            do! saveProject project true checker
```

Comment:
Did this not work properly? It might still be useful to re-save the project into a testing directory, so when you make changes to it they don't remain after the test. Also less code branching.

Reply:
I very much don't want anything touching my original project file.

Reply:
Oh I see. `ProjectDir` should not be set to the actual one, rather the test one - what you get if you use `SyntheticProject.Create`.

Reply:
Pretty sure that it needs to be the real `ProjectDir`.
`checkFile` will eventually use

https://github.com/dotnet/fsharp/blob/2f3ed2119248e80710c5ed16229dc94b177b9f87/tests/FSharp.Test.Utilities/ProjectGeneration.fs#L357

Reply:
Yeah but that should be fine if it saves the file contents there first. 

Maybe there are some dependencies on the actual path. But might still be worth trying out.

Reply:
It does work for the non-transparent compiler when I don't use the actual paths:
![image](https://github.com/dotnet/fsharp/assets/2621499/bee6fb2c-81a8-4869-aa2b-9151443914b2)

Reply:
But did you enable saving them? If it saves them and then says this, there must be a mismatch in the path somewhere. Since it works for the synthetic projects.

I'm just thinking if there is some external issue, like the source code referencing something using `__SOURCE_DIRECTORY__` and then we won't find that thing. Although that would probably have to be a type provider for it to matter for type checks.

Reply:
But it's not a big deal if it works now and you don't want to spend more time on it. We can always improve it later.

Reply:
Yeah, I'd like to revisit these open topics in the future.
Right now, having this merged on the main branch means I can test it out on some more projects.